test/op_shape_test.cpp

```diff
@@ -2305,6 +2305,40 @@ TEST_CASE(slice_shape)
                  input);
 }
 
+TEST_CASE(slice_dyn_shape)
+{
+    migraphx::shape input{migraphx::shape::int32_type, {{2, 2, 0}, {7, 7, 0}, {2, 2, 0}}};
+
+    // Slice axis 1 to size 4-1=3
+    expect_shape(migraphx::shape{migraphx::shape::int32_type, {{2, 2, 0}, {3, 3, 0}, {2, 2, 0}}},
+                 migraphx::make_op("slice", {{"axes", {1}}, {"starts", {1}}, {"ends", {4}}}),
+                 input);
+
+    // Slice multiple axes:  axis 0 to size 2-1=1 and axis 1 to size 4-1=3
+    expect_shape(
+        migraphx::shape{migraphx::shape::int32_type, {{1, 1, 0}, {3, 3, 0}, {2, 2, 0}}},
+        migraphx::make_op("slice", {{"axes", {0, 1}}, {"starts", {1, 1}}, {"ends", {2, 4}}}),
+        input);
+
+    // Todo:  When support is added for non-fixed inputs, test  +/- start/end; clipping values too
+    // large or small;
+    //    what happens if min is less than slice size?
+
+    // // Slice axis 1 from dyn. range 2..7 to range 2..3
+    // // Non-fixed input leads to either fixed shape or clipping; shouldn't change the other axes
+    // input = {migraphx::shape::int32_type, {{2, 5, 3}, {2, 7, 3}, {2, 4, 0}}};
+    // expect_shape(migraphx::shape{migraphx::shape::int32_type, {{2, 5, 3}, {2, 3, 3}, {2, 4, 0}}},
+    //              migraphx::make_op("slice", {{"axes", {1}}, {"starts", {1}}, {"ends", {4}}}),
+    //              input);
```

Comment:
I think I commented this out because even though it basically works, I haven't coded all the out-of-bounds and special case handling.  I added a check that disallows non-fixed shapes for now.